packages/core/src/BaseNavigationContainer.tsx

```diff
@@ -174,18 +198,26 @@ export const BaseNavigationContainer = React.forwardRef(
     });
 
     const getCurrentRoute = useLatestCallback(() => {
-      const state = getRootState();
+      const rootState = getRootState();
 
-      if (state == null) {
+      if (rootState == null) {
         return undefined;
       }
 
-      const route = findFocusedRoute(state);
+      const route = findFocusedRoute(rootState);
 
       return route as Route<string> | undefined;
     });
 
-    const isReady = useLatestCallback(() => listeners.focus[0] != null);
+    const isReady = useLatestCallback(() => {
+      const ready = listeners.focus[0] != null;
+
+      if (ready && !isReadyRef.current) {
+        isReadyRef.current = true;
+      }
+
+      return ready;
+    });
```

Comment:
This is a get function - it shouldn't be modifying values. `isReadyRef` also shouldn't be relying on some checking `isReady` first.